plugins/base/src/main/kotlin/DokkaBase.kt

```diff
@@ -207,4 +208,8 @@ class DokkaBase : DokkaPlugin() {
             MultimodulePageCreator(it)
         }
     }
+
+    val baseSearchbarDataInstaller by extending {
```

Comment:
It's acceptable solution, but my personal preference would be to have this as yet another htmlPreprocessor, which adds to pages tree a new, special page, let's call it `PlainJsPage`, with contents from `searchbarDataInstaller.generatePagesList()` and renders it in HtmlRenderer as `outputWriter.write("scripts/pages", (page as PlainJsPage).content, ".js")`

This omits the need of having another extension point for this particular reason and is more consistent with how we generate similar pages (Index, Navigation, etc)